File: backend/uma/src/main/java/it/tndigitale/a4g/uma/dto/consumi/ConsuntivoDto.java
package it.tndigitale.a4g.uma.dto.consumi;

import java.util.List;

import io.swagger.v3.oas.annotations.media.Schema;
import it.tndigitale.a4g.uma.business.persistence.entity.MotivazioneConsuntivo;
import it.tndigitale.a4g.uma.business.persistence.entity.TipoCarburanteConsuntivo;
import it.tndigitale.a4g.uma.business.persistence.entity.TipoConsuntivo;

@Schema
public class ConsuntivoDto {

	private Long id;
	private TipoConsuntivo tipo;
	private TipoCarburanteConsuntivo carburante;
	private Integer quantita;
	private MotivazioneConsuntivo motivazione;
	private List<InfoAllegatoConsuntivoDto> infoAllegati;

	public Long getId() {
		return id;
	}
	public ConsuntivoDto setId(Long id) {
		this.id = id;
		return this;
	}
	public TipoConsuntivo getTipo() {
		return tipo;
	}
	public ConsuntivoDto setTipo(TipoConsuntivo tipo) {
		this.tipo = tipo;
		return this;
	}
	public TipoCarburanteConsuntivo getCarburante() {
		return carburante;
	}
	public ConsuntivoDto setCarburante(TipoCarburanteConsuntivo carburante) {
		this.carburante = carburante;
		return this;
	}
	public Integer getQuantita() {
		return quantita;
	}
	public ConsuntivoDto setQuantita(Integer quantita) {
		this.quantita = quantita;
		return this;
	}
	public MotivazioneConsuntivo getMotivazione() {
		return motivazione;
	}
	public ConsuntivoDto setMotivazione(MotivazioneConsuntivo motivazione) {
		this.motivazione = motivazione;
		return this;
	}
	public List<InfoAllegatoConsuntivoDto> getInfoAllegati() {
		return infoAllegati;
	}
	public ConsuntivoDto setInfoAllegati(List<InfoAllegatoConsuntivoDto> infoAllegati) {
		this.infoAllegati = infoAllegati;
		return this;
	}
	public ConsuntivoDto addInfoAllegato(InfoAllegatoConsuntivoDto infoAllegato) {
		this.infoAllegati.add(infoAllegato);
		return this;
	}
}

File: frontend/src/app/uma/core-uma/models/dto/ConsuntivoDto.ts
import { MotivazioneConsuntivo } from "../enums/MotivazioneConsuntivo.enum";
import { TipoAllegatoConsuntivo } from "../../../../a4g-common/classi/enums/uma/TipoAllegatoConsuntivo.enum";
import { TipoCarburanteConsuntivo } from "../enums/TipoCarburanteConsuntivo.enum";
import { TipoConsuntivo } from "../enums/TipoConsuntivo";

export class ConsuntivoDto {
    id: string;
    tipo: keyof typeof TipoConsuntivo;
    carburante: keyof typeof TipoCarburanteConsuntivo;
    quantita: number;
    motivazione?: keyof typeof MotivazioneConsuntivo;
    infoAllegati?: Array<InfoAllegatoConsuntivoDto>; 
}

export class InfoAllegatoConsuntivoDto {
    id: string;
    nome: string;
    descrizione: string;
    tipoDocumento: TipoAllegatoConsuntivo;
}